phpBB/phpbb/template/twig/extension/icon.php

```diff
@@ -110,6 +133,20 @@ public function icon(environment $environment, $type, $icon, $title = '', $hidde
 			break;
 		}
 
+		// If no PNG or SVG icon was found, display a default 404 PNG icon.
+		if ($not_found)
+		{
+			if (empty($base_path))
+			{
+				$board_url = defined('PHPBB_USE_BOARD_URL_PATH') && PHPBB_USE_BOARD_URL_PATH;
+				$base_path = $board_url ? generate_board_url() . '/' : $environment->get_web_root_path();
+			}
+
+			$source = "{$base_path}styles/chameleon/theme/icons/png/404.png";
+			$type = 'png';
```

Comment:
can we make this dynamic. check for 404.svg if none then use 404.png.

Reply:
I thought the idea was to provide an icon/image that is always there. So there shouldn't be a need to verify existance of the 'not found icon'. So not sure what you want here? Either display a `404.svg` if the original type was `svg` or what?

Reply:
while we may provide 1 for both this could be customized by someone so we need to check both folders. if there is an 404.svg display it. if not display the 404.png. It will still only display the same thing.

Reply:
well, this changes things slightly as that would mean we would have to verify the existance of the icons and otherwise still end up with no icon.
Isn't more logical to provide a 404.svg and a 404.png in the `all` folder, which should just always be there as backup. 
What's customized are the styles, and they should just provide their own icons and when designed correctly it should never even show these 404 icons.
I thought the 404 icon was intended as an ultimate backup.

Reply:
sorry guess i was getting confused a little by the theme folder. to simplify lets just use the png for everything as originally discussed that way it works on any browser/system.